elastic/kibana
# [ResponseOps][Alerts] Migrate alerts fields fetching to TanStack Query

## Summary

Migrates the `useFetchBrowserFieldCapabilities` hook (renamed to `useFetchAlertsFieldsQuery`) to TanStack Query, following [this organizational logic](https://github.com/elastic/kibana/issues/186448#issuecomment-2228853337).

This PR focuses mainly on the fetching logic itself, leaving the surrounding API surface mostly unchanged since it will be likely addressed in subsequent PRs.

## To verify

1. Create rules that fire alerts in different solutions
2. Check that the alerts table usages work correctly ({O11y, Security, Stack} alerts and rule details pages, ...)
    1. Check that the alerts displayed in the table are coherent with the solution, KQL query, time filter, pagination
    2. Check that pagination changes are reflected in the table
    3. Check that changing the query when in pages > 0 resets the pagination to the first page
    4. Check that the fields browser shows and works correctly (`Fields` button in the alerts table header)

Closes point 2 of https://github.com/elastic/kibana/issues/186448

### Checklist
